orte/mca/plm/rsh/plm_rsh_module.c

```diff
@@ -433,6 +446,19 @@ static int setup_launch(int *argcptr, char ***argvptr,
         orted_cmd = opal_argv_join_range(orted_argv, orted_index, opal_argv_count(orted_argv), ' ');
     }
     opal_argv_free(orted_argv);  /* done with this */
+
+    /* if the user specified a library path to pass, set it up now */
+    param = opal_basename(opal_install_dirs.libdir);
+    if (NULL != mca_plm_rsh_component.pass_libpath) {
+        if (NULL != prefix_dir) {
+            asprintf(&lib_base, "%s:%s/%s", mca_plm_rsh_component.pass_libpath, prefix_dir, param);
+        } else {
+            asprintf(&lib_base, "%s:%s", mca_plm_rsh_component.pass_libpath, param);
```

Comment:
Sure - I'm available at 7am Pacific tomorrow, or right after the OMPI
telecon.


On Mon, Apr 20, 2015 at 4:24 PM, Jeff Squyres <notifications@github.com>
wrote:

> In orte/mca/plm/rsh/plm_rsh_module.c
> <https://github.com/open-mpi/ompi-release/pull/243#discussion_r28738502>:
>
> > @@ -433,6 +446,19 @@ static int setup_launch(int *argcptr, char ***argvptr,
> >          orted_cmd = opal_argv_join_range(orted_argv, orted_index, opal_argv_count(orted_argv), ' ');
> >      }
> >      opal_argv_free(orted_argv);  /* done with this */
> > +
> > +    /* if the user specified a library path to pass, set it up now */
> > +    param = opal_basename(opal_install_dirs.libdir);
> > +    if (NULL != mca_plm_rsh_component.pass_libpath) {
> > +        if (NULL != prefix_dir) {
> > +            asprintf(&lib_base, "%s:%s/%s", mca_plm_rsh_component.pass_libpath, prefix_dir, param);
> > +        } else {
> > +            asprintf(&lib_base, "%s:%s", mca_plm_rsh_component.pass_libpath, param);
>
> If prefix_dir is NULL, shouldn't you add the entire
> opal_install_dirs.libdir (not just the param, which is the basename)? It
> seems odd that all of the libdir is dropped except the basename -- the MCA
> param description doesn't describe this value this way:
>
> Thinking about this more, the interaction of --prefix and this libdir
> param is very... odd. I see how you got here, but it still feels... weird.
>
> Want to chat on the phone about some options and/or rationale here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/open-mpi/ompi-release/pull/243/files#r28738502>.
>